model/src/main/kotlin/DependencyNavigator.kt

```diff
@@ -157,14 +157,7 @@ interface DependencyNavigator {
      */
     private fun getShortestPathForScope(project: Project, scope: String): Map<Identifier, List<Identifier>> {
         val directDependencies = directDependencies(project, scope)
-        val shortestPaths = getShortestPathsForScope(directDependencies)
-
-        val unvisitedDependencies = scopeDependencies(project, scope) - shortestPaths.keys
-        require(unvisitedDependencies.isEmpty()) {
```

Comment:
> This implies that the check cannot fail.

And in any case this should have been `check()` instead of `require()`, just as a side note.